account/forms.py

```diff
@@ -0,0 +1,29 @@
+from django.contrib.auth.forms import UserCreationForm
+from django import forms
+from .models import Profile
+from django.db.transaction import atomic
+
+
+class UserCreationFormCustom(UserCreationForm):
+
+    class Meta(UserCreationForm.Meta):
+        fields = ['first_name', 'last_name', 'username']
+
+    nick = forms.CharField(label='Enter nickname', widget=forms.Textarea, min_length=2, max_length=10)
+    biography = forms.CharField(label='Tell us your story', widget=forms.Textarea, min_length=5)
+
+    @atomic
+    def save(self, commit=True):
+        result = super().save(commit)
+        nick = self.cleaned_data['nick']
+        biography = self.cleaned_data['biography']
+        profile = Profile(nick=nick, biography=biography, user=result)
```

Comment:
skoro używasz sygnałów, to tutaj nie należy drugi raz tworzyc profilu. On już istnieje. Trzeba wziąc taki stworzony profil i zrobic na nim update